config/config.go

```diff
@@ -526,6 +528,24 @@ func ConfigureDirectories() error {
 		return err
 	}
 
+	log.WithField("filepath", "/etc/pterodactyl/passwd").Debug("ensuring passwd file exists")
+	if passwd, err := os.Create("/etc/pterodactyl/passwd"); err != nil {
+		return err
+	} else {
+		shell := "/usr/sbin/nologin"
+		if _config.System.User.Login {
+			shell = "/bin/sh"
+		}
+
+		// the WriteFile method returns an error if unsuccessful
+		err := os.WriteFile(passwd.Name(), []byte(fmt.Sprintf("container:x:%d:%d::/home/container:%s", _config.System.User.Uid, _config.System.User.Gid, shell)), 0777)
```

Comment:
File permission should be `0644`, there is no need for execute permissions on this.